features/backup_restore.py

```diff
@@ -6,6 +6,7 @@
     parser = argparse.ArgumentParser()
     parser.add_argument("--datadir", required=True)
     parser.add_argument("--sourcedir", required=True)
+    parser.add_argument("--test-argument", required=True)
```

Comment:
It is also used for custom replica creation: https://github.com/zalando/patroni/blob/f5ee67fa1c683cacfcc398375e1c55d6a8839c21/features/environment.py#L930-L935
This is the reason why behave tests are failing.

Reply:
🤦 I didn't notice the other behave tests were failing. Thanks for the heads up!

Reply:
Standby cluster behave tests are still failing. In logs I've found the following error: 
```
usage: backup_restore.py [-h] --datadir DATADIR --sourcedir SOURCEDIR
                         --test-argument TEST_ARGUMENT
backup_restore.py: error: the following arguments are required: --test-argument
2023-10-24 16:05:36,677 ERROR [/home/runner/work/patroni/patroni/patroni/postgresql/bootstrap.py:317 - create_replica]: Error creating replica using method backup_restore: /opt/hostedtoolcache/Python/3.7.17/x64/bin/python features/backup_restore.py --sourcedir=/home/runner/work/patroni/patroni/data/basebackup exited with code=2
```
There is at least one more place left: https://github.com/zalando/patroni/blob/f5ee67fa1c683cacfcc398375e1c55d6a8839c21/features/steps/standby_cluster.py#L18-L20

From the first look in all three places there is exactly the same thing happening, the `command` is built from from some class variables in `PatroniPoolController` and from `PatroniPoolController.patroni_path`. But, right now I'm not sure why it is done a bit differently in two places. It could be related to some Windows quirks (need to double check). Although... the thing it does is `.replace('\\', '/')`, which is hard to screw up.

IMO, we should create a new method that will return the entire dict.
something like:
```python
def backup_restore_config(self, **kwargs):
  return {
    'command': (self.BACKUP_RESTORE_SCRIPT + ' --sourcedir=' + os.path.join(self.patroni_path, 'data', 'basebackup')).replace('\\', '/'),
    'test-argument': 'test-value',
    **kwargs
  }
```

and call it from appropriate places.

Reply:
Let's see how it goes, I just submitted a commit. I replaced `**kwargs` with a `params={}` to allow passing things like `test-argument`, which would be an invalid parameter name.